### PR TITLE
Logging Module + Dynamic Buy/Sell Prop

### PR DESCRIPTION
**Pull request by jbdelmundo**
### Description

Added the following changes:
- Docstrings in `BaseStrategy`
- Added a new logging module instead of simply doing `print()` inside the `BaseStrategy`
- Logging inside the strategy can be accessed using `self.logging.log()`
- Separated the function of adding periodic cash
- `buy_signal()` and `sell_signal()` can now return `(signal, buy/sell_prop)` instead of just boolean (still backwards compatible)
- Added `trade_history`
- Set default signal to `False` (no action)

### Checklist
 - [ X] I am making a pull request from a branch other than master
 - [ X] I have read the CONTRIBUTING.md
 - [ X] I have added/edited documentation in a relevant docs/docusaurus markdown file
 - [X ] (For new docs, check if not applicable) I have added the id of a new docs md to the docs/docusaurus/sidebars.js file

### CODE REVIEW

**Comment by jbdelmundo** [comment]
This might be a large change so comments, requests for change are welcome! :)

**Review by rafmacalaba — COMMENTED** [review]
lgtm

**Review by rafmacalaba — APPROVED** [review]

**Comment by jgendrinal** [comment]
*has no idea what I'm doing* lgtm!! 😅 I'll defer to your judgement

File: python/fastquant/strategies/base.py
Comment (enzoampil) [inline python/fastquant/strategies/base.py:42]: These are beautiful docs, thanks so much for this!

File: python/fastquant/strategies/base.py
Comment (enzoampil) [inline python/fastquant/strategies/base.py:74]: Agree with the "internal" renaming, for increased clarity.

File: python/fastquant/strategies/base.py
Comment (enzoampil) [inline python/fastquant/strategies/base.py:67]:
```suggestion
    add_cash_frequency : str {`W`, `M`, or custom cron notion} (default=`W`)
```

Comment (enzoampil) [reply python/fastquant/strategies/base.py:67]: Just added clarification that any custom cron notation can be used to indicate cash inflow frequency / timing

File: python/fastquant/strategies/base.py
Comment (enzoampil) [inline python/fastquant/strategies/base.py:173]: Looks good

File: python/fastquant/strategies/base.py
Comment (enzoampil) [inline python/fastquant/strategies/base.py:57]: @jbdelmundo Any specific reason why you decided to leave out the `open` option that was previously implemented? :smile:

File: python/fastquant/strategies/base.py
Comment (enzoampil) [inline python/fastquant/strategies/base.py:359]: @jbdelmundo  Yeah, I think best to still include the `open` type of execution. Technically, next day open is the most recommended approach to implementing the buy and sell; although, we use current / next day close as a proxy when next day open is not available.

File: python/fastquant/strategies/base.py
Comment (enzoampil) [inline python/fastquant/strategies/base.py:423]: @jbdelmundo I notice that for `create_buy_order` the execution types are `market`, and `close`, while for `create_short_sell_order`, you stuck to the original `close`, and `open` execution types. Is there a reason for why they're not consistent?

**Review by enzoampil — CHANGES_REQUESTED** [review]
Hey @jbdelmundo , thanks so much for this PR. `BaseStrategy` looks 5 times cleaner, and the buy / sell proportions look very interesting :smile:

Let's start with the initial comments that I just gave, most importantly regarding the new "execution types" being implemented, and the inconsistency between the types used for `create_buy_order` vs `create_sell_order`.

My recommendation is to include all three execution types for both:

1. `close` - next time step close
2. `open` - next time step open (this is the preferred `execution_type` though requires an `open` column)
3. `market` - current time step close

*Note: There's current handling that makes `open` the same as `market` in the case that `open` is chosen, but the column doesn't exist in the input dateframe.*

**Review by jgendrinal — APPROVED** [review]

File: python/fastquant/strategies/base.py
Comment (jbdelmundo) [reply python/fastquant/strategies/base.py:57]: Actually I am having trouble making the backtest use the next opening price. 
I have tried using Market and Close orders in backtrader but it seems both use the closing price.
For Close orders, it uses the next day (or bar) close price, and for Market orders, it uses the current closing price

I am using the Ternary Strategy to manually provide buy and sell signals and review them on the order history

**Comment by jbdelmundo** [comment]
Reopened [here](https://github.com/enzoampil/fastquant/pull/380).
I was renaming my remote branch and got deleted

**Pull request closed without merging**
